src/daemon/rpmostreed-transaction-types.cxx

```diff
@@ -1507,6 +1514,47 @@ deploy_transaction_execute (RpmostreedTransaction *transaction,
             changed = TRUE;
         }
     }
+if (override_replace_pkgs)
```

Comment:
So roughly, the way this would work I think would be that in `deploy_transaction_execute`, we just add to the origin the array of `(source, pkgnames[])` elements we got from the client (so we need to think about how we'll represent this in a keyfile). Then in the sysroot upgrader, it'll convert the origin to a treefile to pass to the core: https://github.com/coreos/rpm-ostree/blob/54f3dca04a37671b64a4206008e3fff41cf9e685/src/daemon/rpmostree-sysroot-upgrader.cxx#L963-L967

In `origin_to_treefile_inner`, for the repo case (which is the only source we're starting with for now), we can just map that to `repo-packages`: https://github.com/coreos/rpm-ostree/blob/54f3dca04a37671b64a4206008e3fff41cf9e685/docs/treefile.md#L77-L80

And the core already knows how to handle that (though we might need to tweak bits which assume it's a compose-only option).

Reply:
Ok, thanks, sounds good.